packages/runtime-dom/src/patchProp.ts

```diff
@@ -93,6 +93,12 @@ function shouldSetAsProp(
     return false
   }
 
+  // #1787 form as an attribute must be a string, while it accepts an Element as
+  // a prop
+  if (key === 'form' && typeof value === 'string') {
```

Comment:
I don't know if the type check on value is worth here as setting the `form` is ignored anyway since it's a read only property